cdap-watchdog/src/main/java/co/cask/cdap/metrics/store/timeseries/FactTable.java

```diff
@@ -246,31 +256,32 @@ public int compare(TagValue t1, TagValue t2) {
     byte[] endRow = codec.createEndRowKey(tags, null, endTs, true);
     endRow = Bytes.stopKeyForPrefix(endRow);
     FuzzyRowFilter fuzzyRowFilter = createFuzzyRowFilter(new FactScan(startTs, endTs, null, tags),
-                                                                startRow, true);
+                                                         startRow, true);
     Row rowResult;
     SortedSet<String> measureNames = Sets.newTreeSet();
     Scanner scanner = null;
     long startTimeBase = startTs / rollTime * rollTime;
     long endTimeBase = endTs / rollTime * rollTime;
     try {
-        scanner = timeSeriesTable.scan(startRow, endRow, null, fuzzyRowFilter);
-        while ((rowResult = scanner.next()) != null) {
-          byte[] rowKey = rowResult.getRow();
-          // since the timestamp filter is fuzzy and also the aggregate_key is at the begging of rowKey,
-          // we could match rows where (ts < startTs or ts > endTs) and the tags at beginning match,
-          // we will skip those rows
-          if ((codec.getTimestamp(rowKey, codec.createColumn(startTs)) < startTimeBase) ||
-            (codec.getTimestamp(rowKey, codec.createColumn(endTs)) > endTimeBase)) {
-            continue;
-          }
-          // if the timestamp is within query range and the tag-list matches, we add them to the result
-          if (startsWithTags(tags, codec.getTagValues(rowKey))) {
-            //add item to result set and increment the start rowKey
-            measureNames.add(codec.getMeasureName(rowResult.getRow()));
-            startRow = codec.getNextRowKey(rowResult.getRow(), tags.size());
-            scanner = timeSeriesTable.scan(startRow, endRow, null, fuzzyRowFilter);
-          }
+      scanner = timeSeriesTable.scan(startRow, endRow, null, fuzzyRowFilter);
```

Comment:
Let's add a todo in the very beginning of the method to only go over the aggregations that we know that have the given tags as first tags in same order. No need to implement now, but let's leave a todo